packages/react/src/evaluation/use-feature-flag.ts

```diff
@@ -336,3 +398,52 @@ function suspenseWrapper<T>(promise: Promise<T>) {
     }
   };
 }
+
+// FlagQuery implementation, do not export
+class HookFlagQuery<T extends FlagValue = FlagValue> implements FlagQuery {
+  constructor(private _details: EvaluationDetails<T>) {}
+
+  get details() {
+    return this._details;
+  }
+
+  get value() {
+    return this._details?.value;
+  }
+
+  get variant() {
+    return this._details.variant;
+  }
+
+  get flagMetadata() {
+    return this._details.flagMetadata;
+  }
+
+  get reason() {
+    return this._details.reason;
+  }
+
+  get isError() {
+    return !!this._details?.errorCode || this._details.reason == StandardResolutionReasons.ERROR;
+  }
+
+  get errorCode() {
+    return this._details?.errorCode;
+  }
+
+  get errorMessage() {
+    return this._details?.errorMessage;
+  }
+
+  get isAuthoritative() {
+    return (
+      !this.isError &&
+      this._details.reason != StandardResolutionReasons.STALE &&
+      this._details.reason != StandardResolutionReasons.DISABLED
+    );
+  }
```

Comment:
@moredip this naming makes sense to me, since we specifically say in the spec, that STALE means:

> The resolved value is **_non-authoritative_** or possibly out of date

(emphasis mine).